### PR TITLE
enable more PL integrations

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Fixes #\<issue_number>

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@SkafteNicki this looks strange, how it can pass list but does not have `len()`?
```
            if isinstance(value, list):
>               value = torch.tensor(value)
E               TypeError: object of type 'SumMetric' has no len()
```

**Review by ethanwharris — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: integrations/test_lightning.py [hunk 1]
```diff
@@ -201,45 +201,84 @@ def _assert_called(model, stage):
...
+        def training_epoch_end(self, outs):
+            self.log("sum_epoch", self.metric_epoch(torch.stack([o["data"] for o in outs]).sum()))
+
+    model = TestModel()
+    model.val_dataloader = None
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        limit_train_batches=2,
+        limit_val_batches=2,
```
Comment (rohitgr7) [inline integrations/test_lightning.py:233]:
```suggestion

    trainer = Trainer(
        default_root_dir=tmpdir,
        limit_train_batches=2,
        limit_val_batches=0,
```

File: integrations/test_lightning.py [hunk 2]
```diff
@@ -201,45 +201,84 @@ def _assert_called(model, stage):
...
+        max_epochs=2,
+        log_every_n_steps=1,
+        weights_summary=None,
```
Comment (rohitgr7) [inline integrations/test_lightning.py:236]: Do `metrics` tests not run with pl_master?
I think this flag is deprecated.

File: integrations/test_lightning.py [hunk 3]
```diff
@@ -201,45 +201,84 @@ def _assert_called(model, stage):
...
+            metric_vals = self.metric.compute()
+            self.log_dict({f"{k}_epoch": v for k, v in metric_vals.items()})
+
+    model = TestModel()
+    model.val_dataloader = None
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        limit_train_batches=2,
+        limit_val_batches=2,
```
Comment (rohitgr7) [inline integrations/test_lightning.py:271]:
```suggestion

    trainer = Trainer(
        default_root_dir=tmpdir,
        limit_train_batches=2,
        limit_val_batches=0,
```

**Review by rohitgr7 — APPROVED** [review]

File: torchmetrics/text/bert.py [hunk 4]
```diff
@@ -17,14 +17,14 @@
 import torch
 from deprecate import deprecated
 
-from torchmetrics.functional import bert_score
+from torchmetrics.functional.text import bert_score
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:20]:
```suggestion
from torchmetrics.functional.text.bert import bert_score
```

File: integrations/test_lightning.py (hunk 2 above)
Comment (SkafteNicki) [reply integrations/test_lightning.py:235]:
```suggestion
```

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**